### PR TITLE
Smoothing rebased2

### PR DESCRIPTION
**Pull request by baumanta**
velocity dependent minimum jerk smoothing of waypoint

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Is this SITL tested and ready for flight testing or already flown? Please try to include SITL and/or flight test logs when raising PRs. Thanks!

File: local_planner/src/nodes/waypoint_generator.cpp [hunk 1]
```diff
@@ -246,31 +246,35 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
...
-      (last_position_waypoint_.pose.position.y -
-       last_last_position_waypoint_.pose.position.y) /
-          dt);
-  Eigen::Vector2f acc_waypt_xy((vel_waypt_xy - vel_waypt_xy_prev) / dt);
-
-  if (acc_waypt_xy.norm() > (acc_waypt_xy.norm() / 2.0f)) {
-    vel_xy = (acc_waypt_xy.norm() / 2.0f) * acc_waypt_xy.normalized() * dt +
-             vel_waypt_xy_prev;
+
+  Eigen::Vector2f v_sp = vel_waypt_xy;
```
Comment (mrivi) [inline local_planner/src/nodes/waypoint_generator.cpp]: `vel_wypt_xy` is copied into `v_sp` and never used again.

File: local_planner/src/nodes/waypoint_generator.cpp [hunk 2]
```diff
@@ -246,31 +246,35 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
...
+  Eigen::Vector2f v = cur_vel_xy;
```
Comment (mrivi) [inline local_planner/src/nodes/waypoint_generator.cpp]: same here

File: local_planner/src/nodes/waypoint_generator.cpp [hunk 3]
```diff
@@ -246,31 +246,35 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
...
+  Eigen::Vector2f acc_sp = (v_sp-v)/dt;
```
Comment (mrivi) [inline local_planner/src/nodes/waypoint_generator.cpp]: format

File: local_planner/src/nodes/waypoint_generator.cpp [hunk 4]
```diff
@@ -387,11 +388,8 @@ void WaypointGenerator::getPathMsg() {
     ROS_DEBUG("[WG] pose altitude func: [%f %f %f].", pose_.pose.position.x,
               pose_.pose.position.y, pose_.pose.position.z);
   } else {
-    if (!only_yawed_ && !reached_goal_) {
-      smoothWaypoint();
-      new_yaw_ = nextYaw(pose_, output_.smoothed_goto_position);
-      ;
-    }
+    smoothWaypoint();
+//    new_yaw_ = nextYaw(pose_, output_.smoothed_goto_position);
```
Comment (mrivi) [inline local_planner/src/nodes/waypoint_generator.cpp]: remove commented lines

**Comment by mrivi** [comment]
@LorenzMeier as far as I know, this PR has been already tested outside

**Comment by bkueng** [comment]
@mrivi I cleaned up the code since @baumanta is on holidays.

@LorenzMeier yes this was tested on a vehicle. I think one of the logs is here: https://logs.px4.io/plot_app?log=e918560e-c20c-4623-904d-535e539ccdcf

**Comment by baumanta** [comment]
@Lorenz this is the code that ran yesterday morning outside. The video I shared on slack shows the results achieved by this branch.

**Pull request merged**